ruler/app.py

```diff
@@ -0,0 +1,174 @@
+  return variables
+
+
+def get_envvars():
+  env_var_names = [
+    'OIDC_CLIENT_SECRET_KEYNAME',
+    'OIDC_ISSUER',
+    'OIDC_AUTHORIZATION_ENDPOINT',
+    'OIDC_TOKEN_ENDPOINT',
+    'OIDC_USER_INFO_ENDPOINT',
+    'OIDC_CLIENT_ID'
+  ]
+  return get_variables(os.getenv, env_var_names, MISSING_ENVIRONMENT_VARIABLE_MESSAGE)
+
+
+def get_client(service):
+  return boto3.client(service)
+
+
+def get_client_key(key_name):
+  ssm = get_client('ssm')
+  try:
+    response = ssm.get_parameter(
+      Name=key_name,
+      WithDecryption=True)
+  except ClientError as e:
+    client_error_msg = e.response['Error']['Code']
+    exception_msg = f'{MISSING_CLIENT_SECRET_KEY}: key_name={key_name}; {client_error_msg}'
+    raise Exception(exception_msg)
+
+  return response['Parameter']['Value']
+
+
+def next_priority(listener_arn):
+  elbv2 = get_client('elbv2')
+  try:
+    rules = elbv2.describe_rules(ListenerArn=listener_arn).get('Rules')
+  except ClientError as e:
+    client_error_msg = e.response['Error']['Code']
+    exception_msg = f'{ALB_RULES_ACCESS_ERROR}: listener_arn={listener_arn}; {client_error_msg}'
+    raise Exception(exception_msg)
+
+  priorities = sorted([int(rule['Priority']) for rule in rules if rule['Priority'] != 'default'])
+  priority = 1 if not priorities else priorities[-1] + 1
+  log.debug(f'Next priority value = {priority}')
+  return priority
+
+
+@helper.update
+def update(event, context):
+  '''Handles custom resource update events'''
+  logger.info('Received update event:' + json.dumps(event, sort_keys=False))
+  delete(event, context)
+  physical_resource_id = create(event, context)
+  return physical_resource_id
+
+
+@helper.create
+def create(event, context):
+  '''Handles customm resource create events'''
+  log.info('Start ALB rule-making Lambda processing')
+  log.debug('Received create event: ' + json.dumps(event, sort_keys=False))
+
+  # get variables from lambda properties and environment
+  instance_id, target_group_arn, listener_arn = get_properties(event)
+  oidc_client_secret_key_name, oidc_issuer, oidc_auth_endpoint, oidc_token_endpoint, oidc_user_info_endpoint, oidc_client_id  = get_envvars()
+
+  # get oidc client secret from ssm
+  client_secret = get_client_key(oidc_client_secret_key_name)
+
+  # determine priority for the next rule by checking existing alb listener rules
+  priority = next_priority(listener_arn)
+
+  # create the rule
+  elbv2 = get_client('elbv2')
+  try:
+    response = elbv2.create_rule(
+      ListenerArn=listener_arn,
+      Priority=priority,
+      Conditions=[
+        {
+          "Field": "path-pattern",
+          "PathPatternConfig": {
+              "Values": [f"/{instance_id}"]
+          }
+        }
+      ],
+      Actions=[
+        {
+          "Type": "authenticate-oidc",
+          "AuthenticateOidcConfig": {
+            "Issuer": oidc_issuer,
+            "AuthorizationEndpoint": oidc_issuer,
```

Comment:
shouldn't the value for this be `oidc_auth_endpoint`?

Reply:
good catch